AppFact.SerilogOpenSearchSink/SerilogExtensions.cs

```diff
@@ -50,26 +51,79 @@ public static class SerilogExtensions
         string uri,
         string basicAuthUser,
         string basicAuthPassword,
-        string index = "logs", 
+        string index = "logs",
+        int? maxBatchSize = 1000,
+        double tickInSeconds = 1.0,
+        LogEventLevel restrictedToMinimumLevel = LevelAlias.Minimum,
+        LoggingLevelSwitch? levelSwitch = null,
+        bool bypassSsl = false)
+    {
+        return configuration.OpenSearch(new[] { new Uri(uri) },
+            basicAuthUser,
+            basicAuthPassword,
+            index,
+            maxBatchSize,
+            tickInSeconds,
+            restrictedToMinimumLevel,
+            levelSwitch,
+            bypassSsl);
+    }
+
+
+    /// <summary>
+    /// Adds the OpenSearch sink to the logger configuration using the provided parameters
+    /// </summary>
+    /// <param name="configuration"></param>
+    /// <param name="connectionStrings">Will use <see cref="SniffingConnectionPool"/> if several urls are provided and <see cref="SingleNodeConnectionPool "/> if a single one is provided</param>
+    /// <param name="basicAuthUser">username for basic auth</param>
+    /// <param name="basicAuthPassword">password for basic auth</param>
+    /// <param name="index">index where logs are sent to</param>
+    /// <param name="maxBatchSize">how many logs (maximum) should be sent to OpenSearch in one single request/on each tick</param>
+    /// <param name="tickInSeconds">how often logs are taken from the internal queue and sent to OpenSearch</param>
+    /// <param name="restrictedToMinimumLevel">The minimum level for events passed through the sink. Ignored when levelSwitch is specified.</param>
+    /// <param name="levelSwitch">A switch allowing the pass-through minimum level to be changed at runtime.</param>
+    /// <param name="bypassSsl">Bypass OpenSearch node SSL certificate if it's untrusted. Default behaviour for .NET is to throw exception in such cases.</param>
+    /// <returns></returns>
+    static public LoggerConfiguration OpenSearch(this LoggerSinkConfiguration configuration,
+        IEnumerable<Uri> connectionStrings,
```

Comment:
Please use ICollection here, as to avoid multiple enumeration of an IEnumerable can not occur.
Alternatively, if you want to keep IEnumerable, you can force enumeration before you use it.
A bit more on this [here](https://www.jetbrains.com/help/resharper/PossibleMultipleEnumeration.html)

Reply:
It's a warning, not necessary it will enumerate multiple times. In fact it won't, looking at the soruce code. I am using SonarLint and I think it's handling better such cases. ICollection is also fine in this case. Code updated